includes/Core/Storage/Setting_With_Owned_Keys_Trait.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * Trait Google\Site_Kit\Core\Storage\Setting_With_Owned_Keys_Trait
+ *
+ * @package   Google\Site_Kit\Core\Storage
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Storage;
+
+use Google\Site_Kit\Core\Permissions\Permissions;
+
+/**
+ * Trait for a Setting that has owner ID option key.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+trait Setting_With_Owned_Keys_Trait {
+
+	/**
+	 * Registers hooks to determine an owner ID for a module.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array  $keys                   An array of keys to check.
+	 * @param string $owner_id_key Optional. The key to use for storing owner ID setting.
+	 */
+	protected function register_owned_keys( array $keys, $owner_id_key = 'ownerID' ) {
```

Comment:
Would it be possible to implement `get_owned_keys` as a public abstract method here and then override in the respective classes using the trait? Potentially we could simplify the `register_owned_keys` signature to only support the optional `$owner_id_key` argument, and it would rely on calling `get_owned_keys` as implemented by the class.

It would simplify using this trait.

Reply:
Yep, we can do it. Updated.